config/config.go

```diff
@@ -240,45 +240,47 @@ type EnvironmentConfig struct {
+	ID string `yaml:"id" json:"id"`
+	
+	// Base path for this API.
+	BasePath string `yaml:"base_path,omitempty" json:"base_path,omitempty"`
 
-	// Authentication defines the proxy-level authentication requirement
+	// The default authentication requirements for this API.
 	Authentication AuthenticationRequirement `yaml:"authentication,omitempty" json:"authentication,omitempty"`
 
-	// ConsumerAuthorization defines the proxy-level consumer authorization
+	// The default consumer authorization requirements for this API.
 	ConsumerAuthorization ConsumerAuthorization `yaml:"consumer_authorization,omitempty" json:"consumer_authorization,omitempty"`
 
-	// Name of the target server for this proxy.
-	Target string `yaml:"target" json:"target"`
+	// The default value for the `x-apigee-target` header that will be appended to all allowed requests. 
+	TargetID string `yaml:"target" json:"target"`
 
-	// A list of Operations, names of which must be unique within the proxy config.
+	// A list of API Operations, names of which must be unique within the API.
 	Operations []APIOperation `yaml:"operations,omitempty" json:"operations,omitempty"`
 }
 
-// An APIOperation associates a set of rules with a set of request matching
-// settings.
+// An APIOperation associates a set of rules with a set of request matching settings.
 type APIOperation struct {
-	// Name of the operation. Unique within a proxy config.
+	// Name of the API Operation. Unique within a API.
 	Name string `yaml:"name" json:"name"`
 
-	// Authentication defines the operation-level authentication requirement and overrides whatever in the proxy level
+	// The authentication requirements for thie Operation. If specified, this overrides the default AuthenticationRequirement specified at the API level.
 	Authentication AuthenticationRequirement `yaml:"authentication,omitempty" json:"authentication,omitempty"`
 
-	// ConsumerAuthorization defines the operation-level consumer authorization and overrides whatever in the proxy level
+	// The consumer authorization requirement for this Operation. If specified, this overrides the default ConsumerAuthorization specified at the API level.
 	ConsumerAuthorization ConsumerAuthorization `yaml:"consumer_authorization,omitempty" json:"consumer_authorization,omitempty"`
 
-	// HTTP matching rules for this operation. If omitted, this will match all requests.
+	// HTTP matching rules for this Operation. If omitted, this API Operation will match all HTTP requests not matched by another API Operation.
 	HTTPMatches []HTTPMatch `yaml:"http_match,omitempty" json:"http_match,omitempty"`
 
-	// Name of the target server for this operation.
-	Target string `yaml:"target" json:"target"`
+	// The value for the `x-apigee-target` header for this Operation that will be appended to all allowed requests. 
+	TargetID string `yaml:"target,omitempty" json:"target,omitempty"`
 }
 
 // AuthenticationRequirement is the interface defining the authentication requirement.
```

Comment:
JFYI, I did this wrong here and another wrapper is needed. I fixed this in the ongoing work to load the config. So no action required here.